iocore/net/quic/QUICStream.cc

```diff
@@ -409,3 +417,63 @@ QUICStream::largest_offset_sent()
 {
   return this->_remote_flow_controller->current_offset();
 }
+
+bool
+QUICStream::is_valid_fin_offset(QUICOffset last_offset) const
+{
+  // {sector 11.3. Stream Final Offset}
+  // Once a final offset for a stream is known, it cannot change.
+  // If a RST_STREAM or STREAM frame causes the final offset to change for a stream,
+  // an endpoint SHOULD respond with a FINAL_OFFSET_ERROR error (see Section 12).
+  // A receiver SHOULD treat receipt of data at or beyond the final offset as a
+  if (this->_last_offset == UINT64_MAX) {
+    return true;
+  }
+
+  if (this->_last_offset != last_offset) {
+    return false;
+  }
+
+  return true;
+}
+
+QUICErrorCode
+QUICStream::get_error_code() const
```

Comment:
Where is this method called?

Reply:
It seems we missing rst error msg. And we should reserve this message sometimes.

Reply:
I'm not sure how we can use it yet. Let's focus on FIN handling on this PR.